README.md
# sy_flutter_wechat

微信SDK flutter插件，支持Android和iOS

- [x] 微信支付
- [x] 分享文字
- [x] 分享图片
- [x] 分享链接
- [ ] 分享音乐
- [ ] 分享视频
- [ ] 分享小程序

分享图片及链接暂时仅支持 **网络图片** ，    
iOS分享网络图片如果不是 **HTTPS** 的话可能会失败，因为iOS ATS问题。   
微信分享返回的结果仅代表调用微信分享sdk是否成功，不代表用户确实分享出去了，   
微信2018年05月16日调整[分享政策](https://mp.weixin.qq.com/cgi-bin/announce?action=getannouncement&announce_id=11526372695t90Dn&version=&lang=zh_CN)。

### 使用方法

#### Android
无需配置

#### iOS
1. 参考 [微信文档](https://pay.weixin.qq.com/wiki/doc/api/app/app.php?chapter=8_5) 项目设置APPID
1. 修改 Background Modes，勾选以下两项，否则可能会收不到微信回调
![](https://raw.githubusercontent.com/lishuhao/assets/master/sy_flutter_wechat/background_mode.jpg)

### 示例代码

```dart
import 'dart:convert';

import 'package:flutter/material.dart';
import 'package:sy_flutter_wechat/sy_flutter_wechat.dart';

void main() => runApp(new MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => new _MyAppState();
}

class _MyAppState extends State<MyApp> {
  @override
  void initState() {
    super.initState();
    _register();
  }

  _register() async {
    bool result = await SyFlutterWechat.register('wxf9909bde17439ac2');
    print(result);
  }

  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
      home: new Scaffold(
        appBar: new AppBar(
          title: const Text('Plugin example app'),
        ),
        body: new ListView(
          padding: EdgeInsets.all(8.0),
          children: <Widget>[
            RaisedButton(
              child: Text('分享文字'),
              onPressed: () async {
                bool res = await SyFlutterWechat.shareText('hello world',
                    shareType: SyShareType.session);
                print('分享文字：' + res.toString());
              },
            ),
            RaisedButton(
              child: Text('分享图片'),
              onPressed: () async {
                bool res = await SyFlutterWechat.shareImage(
                    'https://avatars0.githubusercontent.com/u/10024776',
                    shareType: SyShareType.timeline);
                print('分享图片：' + res.toString());
              },
            ),
            RaisedButton(
              child: Text('分享网页'),
              onPressed: () async {
                bool res = await SyFlutterWechat.shareWebPage(
                    '标题',
                    '描述',
                    'https://avatars0.githubusercontent.com/u/10024776',
                    'http://www.example.com',
                    shareType: SyShareType.session);
                print('分享网页：' + res.toString());
              },
            ),
            RaisedButton(
              child: Text('支付'),
              onPressed: () async {
                String payInfo =
                    '{"appid":"wxf9909bde17439ac2","partnerid":"1518469211","prepayid":"wx120649521695951d501636f91748325073","package":"Sign=WXPay","noncestr":"1541976592","timestamp":"1541976592","sign":"E760C99A1A981B9A7D8F17B08EF60FCC"}';
                SyPayResult payResult = await SyFlutterWechat.pay(
                    SyPayInfo.fromJson(json.decode(payInfo)));
                print(payResult);
              },
            ),
          ],
        ),
      ),
    );
  }
}
```

####
其它Flutter plugin

- [支付宝](https://github.com/lishuhao/sy_flutter_alipay)
- [Flutter组件库](https://github.com/lishuhao/sy_flutter_widgets)
- [高德定位](https://github.com/lishuhao/sy_flutter_amap)
- [七牛云存储SDK](https://github.com/lishuhao/sy_flutter_qiniu_storage)

CHANGELOG.md

## 0.2.2
update pubspec.yaml description
add more doc

## 0.2.1
podspec add s.static_framework = true  
related [issue](https://github.com/lishuhao/sy_flutter_wechat/issues/1)

## 0.1.3

* initial release.


example/test/widget_test.dart
// This is a basic Flutter widget test.
// To perform an interaction with a widget in your test, use the WidgetTester utility that Flutter
// provides. For example, you can send tap and scroll gestures. You can also use WidgetTester to
// find child widgets in the widget tree, read text, and verify that the values of widget properties
// are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:sy_flutter_wechat_example/main.dart';

void main() {
  testWidgets('Verify Platform version', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(new MyApp());

    // Verify that platform version is retrieved.
    expect(
        find.byWidgetPredicate(
          (Widget widget) =>
              widget is Text && widget.data.startsWith('Running on:'),
        ),
        findsOneWidget);
  });
}


lib/sy_flutter_wechat.dart
import 'dart:async';

import 'package:flutter/services.dart';

/// share scene
enum SyShareType {
  session,

  /// share to friend
  timeline,

  /// share to timeline
  favorite

  /// collect
}

/// pay result
enum SyPayResult {
  /// success
  success,

  /// fail
  fail,

  /// cancel
  cancel,
}

/// wechat class
class SyFlutterWechat {
  static const MethodChannel _channel =
      const MethodChannel('sy_flutter_wechat');

  /// register app id
  static Future<bool> register(String appId) async {
    return await _channel
        .invokeMethod('register', <String, dynamic>{'appId': appId});
  }

  /// shareText
  static Future<bool> shareText(String text, {SyShareType shareType}) async {
    return await _channel.invokeMethod('shareText', <String, dynamic>{
      'text': text,
      'shareType': _shareTypeToStr(shareType)
    });
  }

  /// shareImage
  static Future<bool> shareImage(String imageUrl,
      {SyShareType shareType}) async {
    return await _channel.invokeMethod('shareImage', <String, dynamic>{
      'imageUrl': imageUrl,
      'shareType': _shareTypeToStr(shareType)
    });
  }

  /// shareWebPage
  static Future<bool> shareWebPage(
      String title, String description, String imageUrl, String webPageUrl,
      {SyShareType shareType}) async {
    return await _channel.invokeMethod('shareWebPage', <String, dynamic>{
      'title': title,
      'description': description,
      'imageUrl': imageUrl,
      'webPageUrl': webPageUrl,
      'shareType': _shareTypeToStr(shareType)
    });
  }

  /// pay
  static Future<SyPayResult> pay(SyPayInfo payInfo) async {
    int payResult = await _channel.invokeMethod('pay', <String, dynamic>{
      'appid': payInfo.appid,
      'partnerid': payInfo.partnerid,
      'prepayid': payInfo.prepayid,
      'package': payInfo.package,
      'noncestr': payInfo.noncestr,
      'timestamp': payInfo.timestamp,
      'sign': payInfo.sign,
    });
    return _convertPayResult(payResult);
  }

  static String _shareTypeToStr(SyShareType shareType) {
    switch (shareType) {
      case SyShareType.session:
        return 'session';
      case SyShareType.timeline:
        return 'timeline';
      case SyShareType.favorite:
        return 'favorite';
      default:
        return 'session';
    }
  }

  static SyPayResult _convertPayResult(int payResult) {
    switch (payResult) {
      case 0:
        return SyPayResult.success;
      case -1:
        return SyPayResult.fail;
      case -2:
        return SyPayResult.cancel;
      default:
        return null;
    }
  }
}

/// pay args
///
/// 参考微信文档 https://pay.weixin.qq.com/wiki/doc/api/app/app.php?chapter=9_12&index=2
class SyPayInfo {
  /// appid
  String appid;

  /// partnerid
  String partnerid;

  /// prepayid
  String prepayid;

  /// package
  String package;

  /// noncestr
  String noncestr;

  /// timestamp
  String timestamp;

  /// sign
  String sign;

  SyPayInfo({
    this.appid,
    this.partnerid,
    this.prepayid,
    this.package,
    this.noncestr,
    this.timestamp,
    this.sign,
  });

  /// from json
  factory SyPayInfo.fromJson(Map<String, dynamic> json) {
    return SyPayInfo(
      appid: json['appid'],
      partnerid: json['partnerid'],
      prepayid: json['prepayid'],
      package: json['package'],
      noncestr: json['noncestr'],
      timestamp: json['timestamp'],
      sign: json['sign'],
    );
  }
}


ios/Classes/SyFlutterWechatPlugin.m
#import "SyFlutterWechatPlugin.h"

@interface SyFlutterWechatPlugin()

@property FlutterResult result;

@end

@implementation SyFlutterWechatPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"sy_flutter_wechat"
            binaryMessenger:[registrar messenger]];
  SyFlutterWechatPlugin* instance = [[SyFlutterWechatPlugin alloc] init];
    [registrar addApplicationDelegate:instance];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"register" isEqualToString:call.method]) {
      BOOL res = [WXApi registerApp:call.arguments[@"appId"]];
      result(@((bool)res));
  }else if ([@"shareText" isEqualToString:call.method]) {
      [self shareText:call result:result];
  }else if ([@"shareImage" isEqualToString:call.method]) {
      [self shareImage:call result:result];
  }else if ([@"shareWebPage" isEqualToString:call.method]) {
      [self shareWebPage:call result:result];
  }else if ([@"pay" isEqualToString:call.method]) {
      self.result = result;
      [self pay:call result:result];
  }else {
      result(FlutterMethodNotImplemented);
  }
}


- (void)shareText:(FlutterMethodCall*)call result:(FlutterResult)result{
    SendMessageToWXReq* req = [[SendMessageToWXReq alloc] init];
    req.text = call.arguments[@"text"];
    req.bText = YES;
    req.scene = [self _convertShareType:call];
    BOOL res = [WXApi sendReq:req];
    result(@((bool)res));
}

- (void)shareImage:(FlutterMethodCall*)call result:(FlutterResult)result{
    NSString *imageUrl = call.arguments[@"imageUrl"];
    NSData *imageData =[NSData dataWithContentsOfURL:[NSURL URLWithString:imageUrl]];;
    UIImage *originImage = [UIImage imageWithData:imageData];
    
    WXImageObject *imageObj = [WXImageObject object];
    imageObj.imageData = imageData;
    
    WXMediaMessage *mediaMsg = [WXMediaMessage message];
    mediaMsg.mediaObject = imageObj;
    
    UIImage *thumbImage = [self compressImage:originImage toByte:32768];
    [mediaMsg setThumbImage:thumbImage];
    
    SendMessageToWXReq* req = [[SendMessageToWXReq alloc] init];
    req.message = mediaMsg;
    req.bText = NO;
    req.scene = [self _convertShareType:call];
    BOOL res = [WXApi sendReq:req];
    result(@((bool)res));
}

- (void)shareWebPage:(FlutterMethodCall*)call result:(FlutterResult)result{
    NSString *imageUrl = call.arguments[@"imageUrl"];
    
    NSData *imageData =[NSData dataWithContentsOfURL:[NSURL URLWithString:imageUrl]];;
    UIImage *originImage = [UIImage imageWithData:imageData];
    UIImage *thumbImage = [self compressImage:originImage toByte:32768];
    
    WXWebpageObject* webObj = [WXWebpageObject object];
    webObj.webpageUrl = call.arguments[@"webPageUrl"];

    WXMediaMessage *mediaMsg = [WXMediaMessage message];
    mediaMsg.title = call.arguments[@"title"];
    mediaMsg.description = call.arguments[@"description"];
    [mediaMsg setThumbImage:thumbImage];
    mediaMsg.mediaObject = webObj;
    
    SendMessageToWXReq* req = [[SendMessageToWXReq alloc] init];
    req.message = mediaMsg;
    req.bText = NO;
    req.scene = [self _convertShareType:call];
    BOOL res = [WXApi sendReq:req];
    result(@((bool)res));
}

//支付
- (void)pay:(FlutterMethodCall*)call result:(FlutterResult)result{
    PayReq *req = [[PayReq alloc] init];
    req.partnerId = call.arguments[@"partnerid"];
    req.prepayId= call.arguments[@"prepayid"];
    req.package = call.arguments[@"package"];
    req.nonceStr= call.arguments[@"noncestr"];
    req.timeStamp= [call.arguments[@"timestamp"] unsignedIntValue];
    req.sign= call.arguments[@"sign"];
    [WXApi sendReq:req];
}



- (enum WXScene)_convertShareType:(FlutterMethodCall*)call{
    NSString *shareType = call.arguments[@"shareType"];
    if([shareType isEqualToString:@"session"]){
        return WXSceneSession;
    }else if([shareType isEqualToString:@"timeline"]){
        return WXSceneTimeline;
    }else if([shareType isEqualToString:@"favorite"]){
        return WXSceneFavorite;
    }
    return WXSceneSession;
}


//生成分享缩略图
//链接：https://www.jianshu.com/p/a45d99ffccf6
//toByte 缩略图要小于32KB
- (UIImage *)compressImage:(UIImage *)image toByte:(NSUInteger)maxLength {
    // Compress by quality
    CGFloat compression = 1;
    NSData *data = UIImageJPEGRepresentation(image, compression);
    if (data.length < maxLength) return image;
    
    CGFloat max = 1;
    CGFloat min = 0;
    for (int i = 0; i < 6; ++i) {
        compression = (max + min) / 2;
        data = UIImageJPEGRepresentation(image, compression);
        if (data.length < maxLength * 0.9) {
            min = compression;
        } else if (data.length > maxLength) {
            max = compression;
        } else {
            break;
        }
    }
    UIImage *resultImage = [UIImage imageWithData:data];
    if (data.length < maxLength) return resultImage;
    
    // Compress by size
    NSUInteger lastDataLength = 0;
    while (data.length > maxLength && data.length != lastDataLength) {
        lastDataLength = data.length;
        CGFloat ratio = (CGFloat)maxLength / data.length;
        CGSize size = CGSizeMake((NSUInteger)(resultImage.size.width * sqrtf(ratio)),
                                 (NSUInteger)(resultImage.size.height * sqrtf(ratio))); // Use NSUInteger to prevent white blank
        UIGraphicsBeginImageContext(size);
        [resultImage drawInRect:CGRectMake(0, 0, size.width, size.height)];
        resultImage = UIGraphicsGetImageFromCurrentImageContext();
        UIGraphicsEndImageContext();
        data = UIImageJPEGRepresentation(resultImage, compression);
    }
    
    return resultImage;
}


- (void) onReq:(BaseReq *)req{
    NSLog(@"onReq....");
}

- (void) onResp:(BaseResp *)resp{
    if([resp isKindOfClass:[PayResp class]]){
        self.result(@(resp.errCode));
        NSLog(@"支付结果：retcode = %d, retstr = %@",resp.errCode,resp.errStr);
    }
    NSLog(@"回调结果：retcode = %d, retstr = %@",resp.errCode,resp.errStr);
}

- (BOOL)application:(UIApplication *)application handleOpenURL:(NSURL *)url{
    return [WXApi handleOpenURL:url delegate:self];
}
- (BOOL)application:(UIApplication *)application openURL:(NSURL *)url sourceApplication:(NSString *)sourceApplication annotation:(id)annotation{
    return [WXApi handleOpenURL:url delegate:self];
}
- (BOOL)application:(UIApplication *)application openURL:(NSURL *)url options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options{
    if([url.host isEqualToString:@"pay"]){
        return [WXApi handleOpenURL:url delegate:self];
    }
    return YES;
}

@end


android/src/main/java/plugin/wechat/flutter/isanye/cn/syflutterwechat/SyFlutterWechatPlugin.java
package plugin.wechat.flutter.isanye.cn.syflutterwechat;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.util.Log;

import com.tencent.mm.opensdk.modelmsg.SendMessageToWX;
import com.tencent.mm.opensdk.modelmsg.WXImageObject;
import com.tencent.mm.opensdk.modelmsg.WXMediaMessage;
import com.tencent.mm.opensdk.modelmsg.WXTextObject;
import com.tencent.mm.opensdk.modelmsg.WXWebpageObject;
import com.tencent.mm.opensdk.modelpay.PayReq;
import com.tencent.mm.opensdk.openapi.IWXAPI;
import com.tencent.mm.opensdk.openapi.WXAPIFactory;

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.net.URL;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;
import plugin.wechat.flutter.isanye.cn.syflutterwechat.wxapi.StateManager;

/** SyFlutterWechatPlugin */
public class SyFlutterWechatPlugin implements MethodCallHandler {

  private static final String TAG = "SyFlutterWechatPlugin>>";
  public static final String filterName = "wxCallback";
  private IWXAPI wxApi;
  private Registrar registrar;
  private static Result result;
  private static final int THUMB_SIZE = 150;


  //微信支付回调
  private static BroadcastReceiver wxpayCallbackReceiver = new BroadcastReceiver() {
    @Override
    public void onReceive(Context context, Intent intent) {
      Integer errCode = intent.getIntExtra("errCode",-3);
      Log.e(TAG,errCode.toString());
      result.success(errCode);
    }
  };

  /** Plugin registration. */
  public static void registerWith(Registrar registrar) {
    final MethodChannel channel = new MethodChannel(registrar.messenger(), "sy_flutter_wechat");
    final SyFlutterWechatPlugin plugin = new SyFlutterWechatPlugin(registrar);
    channel.setMethodCallHandler(plugin);
    registrar.context().registerReceiver(wxpayCallbackReceiver,new IntentFilter(filterName));
  }

  private SyFlutterWechatPlugin(Registrar registrar){
    this.registrar = registrar;
  }


  @Override
  public void onMethodCall(MethodCall call, Result result) {
    SyFlutterWechatPlugin.result = result;
    switch (call.method) {
      case "getPlatformVersion":
        result.success("Android " + android.os.Build.VERSION.RELEASE);
        break;
      case "register":
        this.registerToWX(call,result);
        break;
      case "shareText":
        this.shareText(call,result);
        break;
      case "shareImage":
        this.shareImage(call,result);
        break;
      case "shareWebPage":
        this.shareWebPage(call,result);
        break;
      case "pay":
        this.pay(call);
        break;
      default:
        result.notImplemented();
        break;
    }
  }

  //注册微信app id
  private void registerToWX(MethodCall call, Result result){
    String appId = call.argument("appId");
    wxApi = WXAPIFactory.createWXAPI(registrar.context(),appId);
    boolean res =wxApi.registerApp(appId);
    StateManager.setApi(wxApi);
    result.success(res);
  }

  private void shareText(MethodCall call, Result result){
    String text = call.argument("text");
    String shareType = call.argument("shareType");

    WXTextObject textObj = new WXTextObject();
    textObj.text = text;

    WXMediaMessage msg = new WXMediaMessage();
    msg.mediaObject = textObj;
    msg.description = text;

    SendMessageToWX.Req req = new SendMessageToWX.Req();
    req.scene = _convertShareType(shareType);
    req.message = msg;
    //req.transaction = buildTransaction("");
    boolean res = wxApi.sendReq(req);
    //result.success(res);
  }

  private void shareImage(final MethodCall call, final Result result){
    final String imageUrl = call.argument("imageUrl");
    final String shareType = call.argument("shareType");
    new Thread(new Runnable() {
        WXMediaMessage msg = new WXMediaMessage();
        @Override
        public void run() {
          try {
            Bitmap bmp = BitmapFactory.decodeStream(new URL(imageUrl).openStream());
            WXImageObject imageObject = new WXImageObject(bmp);
            Bitmap thumbBmp = Bitmap.createScaledBitmap(bmp, THUMB_SIZE, THUMB_SIZE, true);

            msg.mediaObject = imageObject;
            bmp.recycle();
            msg.thumbData = SyFlutterWechatPlugin.bmpToByteArray(thumbBmp, true);
          }catch (IOException e){
            e.printStackTrace();
          }

          SendMessageToWX.Req req = new SendMessageToWX.Req();
          req.scene = _convertShareType(shareType);
          req.message = msg;
          boolean res = wxApi.sendReq(req);
          result.success(res);
        }
      }
    ).start();
  }

  private void shareWebPage(final MethodCall call, final Result result){
    final String title = call.argument("title");
    final String description = call.argument("description");
    final String imageUrl = call.argument("imageUrl");
    final String webPageUrl = call.argument("webPageUrl");
    final String shareType = call.argument("shareType");
    new Thread(new Runnable() {
      WXWebpageObject webPage = new WXWebpageObject();

      @Override
      public void run() {
        webPage.webpageUrl = webPageUrl;
        WXMediaMessage msg = new WXMediaMessage(webPage);
        msg.title = title;
        msg.description = description;
        try {
          Bitmap bmp = BitmapFactory.decodeStream(new URL(imageUrl).openStream());
          Bitmap thumbBmp = Bitmap.createScaledBitmap(bmp, THUMB_SIZE, THUMB_SIZE, true);
          bmp.recycle();
          msg.thumbData = SyFlutterWechatPlugin.bmpToByteArray(thumbBmp, true);
        }catch (IOException e){
          e.printStackTrace();
        }

        SendMessageToWX.Req req = new SendMessageToWX.Req();
        req.scene = _convertShareType(shareType);
        req.message = msg;
        boolean res = wxApi.sendReq(req);
        result.success(res);
      }
    }
    ).start();
  }

  //调起微信支付
  private void pay(MethodCall call){
    PayReq req = new PayReq();
    req.appId = call.argument("appid");
    req.partnerId = call.argument("partnerid");
    req.prepayId= call.argument("prepayid");
    req.packageValue = call.argument("package");
    req.nonceStr= call.argument("noncestr");
    req.timeStamp= call.argument("timestamp");
    req.sign= call.argument("sign");
    wxApi.sendReq(req);
  }


  private static int _convertShareType(String shareType){
    switch (shareType){
      case "session":
        return SendMessageToWX.Req.WXSceneSession;
      case "timeline":
        return SendMessageToWX.Req.WXSceneTimeline;
      case "favorite":
        return SendMessageToWX.Req.WXSceneFavorite;
      default:
        return SendMessageToWX.Req.WXSceneSession;
    }
  }

  private static byte[] bmpToByteArray(final Bitmap bmp, final boolean needRecycle) {
    ByteArrayOutputStream output = new ByteArrayOutputStream();
    bmp.compress(Bitmap.CompressFormat.PNG, 100, output);
    if (needRecycle) {
      bmp.recycle();
    }

    byte[] result = output.toByteArray();
    try {
      output.close();
    } catch (Exception e) {
      e.printStackTrace();
    }

    return result;
  }

}


android/src/main/java/plugin/wechat/flutter/isanye/cn/syflutterwechat/wxapi/WXPayEntryActivity.java
package plugin.wechat.flutter.isanye.cn.syflutterwechat.wxapi;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import com.tencent.mm.opensdk.constants.ConstantsAPI;
import com.tencent.mm.opensdk.modelbase.BaseReq;
import com.tencent.mm.opensdk.modelbase.BaseResp;
import com.tencent.mm.opensdk.openapi.IWXAPI;
import com.tencent.mm.opensdk.openapi.IWXAPIEventHandler;

import plugin.wechat.flutter.isanye.cn.syflutterwechat.SyFlutterWechatPlugin;

public class WXPayEntryActivity extends Activity implements IWXAPIEventHandler{
    private static final String TAG = "WXPayEntryActivity";
    //微信支付
    private IWXAPI iwxapi;

    @Override
    protected void onCreate( Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        iwxapi = StateManager.getAPi();
        iwxapi.handleIntent(getIntent(), this);
    }

    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);
        setIntent(intent);
        iwxapi.handleIntent(intent, this);
    }

    @Override
    public void onReq(BaseReq baseReq) {
        Log.e(TAG,"req");
    }

    @Override
    public void onResp(BaseResp baseResp) {
        Log.e(TAG,"微信支付回调");

        //微信支付
        if(baseResp.getType() == ConstantsAPI.COMMAND_PAY_BY_WX){
            Log.e(TAG,"errCode:"+String.valueOf(baseResp.errCode));
            Intent i = new Intent(SyFlutterWechatPlugin.filterName);
            i.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            i.putExtra("errCode", baseResp.errCode);
            this.sendBroadcast(i);
        }
        finish();
        Log.e(TAG,baseResp.errStr);
    }
}


example/lib/main.dart
import 'dart:convert';

import 'package:flutter/material.dart';
import 'package:sy_flutter_wechat/sy_flutter_wechat.dart';

void main() => runApp(new MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => new _MyAppState();
}

class _MyAppState extends State<MyApp> {
  @override
  void initState() {
    super.initState();
    _register();
  }

  _register() async {
    bool result = await SyFlutterWechat.register('wxf9909bde17439ac2');
    print(result);
  }

  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
      home: new Scaffold(
        appBar: new AppBar(
          title: const Text('Plugin example app'),
        ),
        body: new ListView(
          padding: EdgeInsets.all(8.0),
          children: <Widget>[
            RaisedButton(
              child: Text('分享文字'),
              onPressed: () async {
                bool res = await SyFlutterWechat.shareText('hello world',
                    shareType: SyShareType.session);
                print('分享文字：' + res.toString());
              },
            ),
            RaisedButton(
              child: Text('分享图片'),
              onPressed: () async {
                bool res = await SyFlutterWechat.shareImage(
                    'https://avatars0.githubusercontent.com/u/10024776',
                    shareType: SyShareType.timeline);
                print('分享图片：' + res.toString());
              },
            ),
            RaisedButton(
              child: Text('分享网页'),
              onPressed: () async {
                bool res = await SyFlutterWechat.shareWebPage(
                    '标题',
                    '描述',
                    'https://avatars0.githubusercontent.com/u/10024776',
                    'http://www.example.com',
                    shareType: SyShareType.session);
                print('分享网页：' + res.toString());
              },
            ),
            RaisedButton(
              child: Text('支付'),
              onPressed: () async {
                String payInfo =
                    '{"appid":"wxf9909bde17439ac2","partnerid":"1518469211","prepayid":"wx120649521695951d501636f91748325073","package":"Sign=WXPay","noncestr":"1541976592","timestamp":"1541976592","sign":"E760C99A1A981B9A7D8F17B08EF60FCC"}';
                SyPayResult payResult = await SyFlutterWechat.pay(
                    SyPayInfo.fromJson(json.decode(payInfo)));
                print(payResult);
              },
            ),
          ],
        ),
      ),
    );
  }
}
